### ISSUE TITLE
Exposing GET handler for health check

### ISSUE THREAD

**Issue by tarrencev** [paritytech/jsonrpsee#762]
Is there a recommended approach to exposing a GET handler for i.e. health checks or metrics on the base server? It doesn't seem possible to register additional handlers today. I has hoping `build_from_hyper` would help with this but it still doesn't give access to the underlying service builder.

Related https://github.com/eqlabs/pathfinder/issues/194

**Comment by niklasad1** [comment]
You referring to the HTTP server correct?

So currently, we have the middleware that you can use to collect metrics based on the timing but not a health API or any mean to register additional handlers.

However, I have nothing against to expose a health API on the HTTP server sounds like a good idea.

If you want to implement it that would be great.

**Comment by niklasad1** [comment]
I think [the jsonrpc crate health API](https://docs.rs/jsonrpc-http-server/latest/jsonrpc_http_server/struct.ServerBuilder.html#method.health_api) looks quite clean.

I'm not exactly sure if how likely it's that folks would want to provide their own HTTP status codes, jsonrpc for instance let you define a RPC call and if that fails it will be converted to `status code 500`. That's quite simple to use at least.

Would be neat to have something a bit more flexible such as (a closure could work too):

```rust
trait HealthApi {
    fn health(&self) -> (StatusCode, Body);
}
```

server impl
```rust
  builder.health_api(set_handler());

 match (request.method(), server.get_health_api()) {
    (handler, Method::GET)  {
       let (status_code, body) = handler.health();
       send_http_response(status_code, body);
    }
 }  
```

**Comment by niklasad1** [comment]
sorry @tarrencev I realized that we need this ASAP too.

So I have opened https://github.com/paritytech/jsonrpsee/pull/763 I hope that will fix things for you, sorry for not letting you implement that.

we could revisit the API at a later point, if something comes up with something more better.

**Comment by niklasad1** [comment]
Closed by #763

**Comment by tarrencev** [comment]
Awesome appreciate you taking care of this!
